packages/dds/matrix/src/matrix.ts

```diff
@@ -624,25 +820,50 @@ export class SharedMatrix<T = any>
 				this.rows.applyMsg(msg, local);
 				break;
 			default: {
+				if (contents.type === MatrixOp.changeSetCellPolicy) {
+					this.setCellPolicySwitchOpSeqNumber = rawMessage.sequenceNumber;
+					return;
+				}
+
 				assert(
 					contents.type === MatrixOp.set,
 					0x021 /* "SharedMatrix message contents have unexpected type!" */,
 				);
 
-				const { row, col } = contents;
-
+				const { row, col, value } = contents;
+				assert(rawMessage.clientId !== null, "clientId should not be null!!");
 				if (local) {
 					// We are receiving the ACK for a local pending set operation.
 					const { rowHandle, colHandle, localSeq } = localOpMetadata as ISetOpMetadata;
-
-					// If this is the most recent write to the cell by the local client, remove our
-					// entry from 'pendingCliSeqs' to resume allowing remote writes.
-					if (this.isLatestPendingWrite(rowHandle, colHandle, localSeq)) {
-						this.pending.setCell(rowHandle, colHandle, undefined);
+					if (this.isSetCellOpMadeInFWWPolicy(rawMessage)) {
+						if (this.shouldSetCellBasedOnFWW(rowHandle, colHandle, rawMessage)) {
+							this.cellLastWriteTracker.setCell(rowHandle, colHandle, {
+								seqNum: rawMessage.sequenceNumber,
+								clientId: rawMessage.clientId,
+							});
+						} else {
+							// conflict. We would already set the right contents in the cell, when we would have received
+							// set op for that cell from other client.
+							this.emit(
+								"conflict",
```

Comment:
Is there any change to refactor / reuse some code? I.e. share between resubmit() flow, between local and non-local branches? Asking mostly due to fear that at some point someone will fix a bug only in one branch and forget about all others.

Reply:
Refactored the resubmit flow after removing the isLatestPendingWrite but not the processCore function as the code is quite different for local/non local op and I think most of the ddses have such kind of separate branches for local/non local ops processing depnding upon the behavour. Plus we have tests too to hopefully catch any wrong changes.